File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nonZero=0;
        for(int zero=0;zero<nums.size();zero++){
            if(nums[zero]!=0){
                swap(nums[zero],nums[nonZero]);
                nonZero++;
            }
        }
    }
};

File: 283-move-zeroes/move-zeroes.java
// import java.util.Collections;
class Solution {
    public void moveZeroes(int[] nums) {
        int nonZero=0;
        int zero=0;
        while(zero<nums.length){
            if(nums[zero]!=0)
            {
                // Swap
                int temp=nums[zero];
                nums[zero]=nums[nonZero];
                nums[nonZero]=temp;
                nonZero++;
            }
            zero++;
        }
    }
}